server/src/main/java/com/arcbees/beeshop/server/guice/DispatchServletModule.java

```diff
@@ -20,5 +17,7 @@ protected void configureServlets() {
         for (String locale : LocaleExtractor.SUPPORTED_LOCALES) {
             serveRegex("/" + locale, "/" + locale + "/").with(RootServlet.class);
         }
+
+        serve("/*").with(NotFoundServlet.class);
```

Comment:
@olafleur This is a server side rendered page GWT/GIN bindings can't be done here. This is due to the fact that because of i18n/SEO, we need to be able serve different `index.html` based on what the user's locale is.